message/pb/message.proto

```diff
@@ -18,6 +18,7 @@ message Message {
 			bool cancel = 3;		// whether this revokes an entry
       WantType wantType = 4; // Note: defaults to enum 0, ie Block
       bool sendDontHave = 5; // Note: defaults to false
+      bytes token = 6;
```

Comment:
It's probably more efficient to group entries by token rather than having each entry contain its own token.

Also, we likely need to key the tokens in some way so the receiver knows what types of tokens they are instead of needing to guess.